### PR TITLE
Add .NET Core SDK image

### PR DESCRIPTION
**Pull request by gnakagaw**
add [.NET Core](https://dotnet.microsoft.com/) image for C# and Visual Basic .NET support

### CODE REVIEW

File: dotnet/Dockerfile.erb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+FROM debian:buster-20200130
+
+<%= ERB.new(File.read('base/Dockerfile.env.erb')).result %>
```
Comment (gnakagaw) [inline dotnet/Dockerfile.erb]: > __[hadolint]__ dotnet/Dockerfile.erb:3:1: ([see on Sider](https://sider.review/gh/repos/200181683/pulls/136#issue-5121310))
>
> <details>
>
>   |
> 3 | <%= ERB.new(File.read('base/Dockerfile.env.erb')).result %>
>   | ^
> unexpected '<'
> expecting '#', ADD, ARG, CMD, COPY, ENTRYPOINT, ENV, EXPOSE, FROM, HEALTHCHECK, LABEL, MAINTAINER, ONBUILD, RUN, SHELL, STOPSIGNAL, USER, VOLUME, WORKDIR, end of input, or whitespace
>
> | Rule | Severity |
> | - | - |
> | `DL1000` | error |
>
> References:
> - <https://github.com/hadolint/hadolint/wiki/DL1000>
>
> </details>

.erbファイルにhadolintが反応しているため無視します。

File: dotnet/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+# set default shell to bash (for pipefail opiton)
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+
+# Disabling .NET Core SDK telemetry
+# https://docs.microsoft.com/en-us/dotnet/core/tools/telemetry
+ENV DOTNET_CLI_TELEMETRY_OPTOUT 1
+
+# install .NET Core SDK
+# https://docs.microsoft.com/en-us/dotnet/core/install/linux-package-manager-debian9
```
Comment (yykamei) [inline dotnet/Dockerfile]: [nitpick] There seems no difference between Debian 9 (Stretch) and Debian 10 (Buster), but we are basically using Buster, so the link below would be more appropriate.

```suggestion
# https://docs.microsoft.com/en-us/dotnet/core/install/linux-package-manager-debian10
```

Comment (gnakagaw) [reply dotnet/Dockerfile]: I removed this link because of using official base image.

**Comment by gnakagaw** [comment]
@yykamei 
Thank you for a kind review.

> Can we use this Docker image? 
It's a reasonable choice. I confirmed that we can use it. (We can use the target static analyzers in the image)

**Comment by ybiquitous** [comment]
Sorry, wait. CI does not run...? 🤔

**Comment by ybiquitous** [comment]
Perhaps we need it:

```diff
-on: [push]
+on: [push, pull_request]
```

https://github.com/sider/devon_rex/blob/f9a57c4ef047d62a9850a450ac07a3c2c55dbc11/.github/workflows/build.yml#L3

**Comment by gnakagaw** [comment]
Thanks all.

**Comment by ybiquitous** [comment]
I've done https://github.com/sider/devon_rex/pull/136#issuecomment-584974959.

@gnakagaw Could you please rebase the master branch?

**Comment by gnakagaw** [comment]
I got many build error. But, I found the cause. Now, I'll fix it.

**Comment by gnakagaw** [comment]
Hmm.. My environment ignored .env value ($LANGUAGE)....

**Comment by gnakagaw** [comment]
fixed it. waiting CI..

**Comment by ybiquitous** [comment]
@gnakagaw You can confirm it like this:

```diff
diff --git a/Rakefile b/Rakefile
index 5fef40e..3206615 100644
--- a/Rakefile
+++ b/Rakefile
@@ -1,6 +1,8 @@
 require 'erb'
 require 'dotenv/load'
 
+pp ENV['LANGUAGE']
+
 ENV['GLOBAL_RUBY_VERSION'] = Pathname('.ruby-version').read.chomp!
 
 BUILD_CONTEXTS = %w[
```

```
$ bundle exec rake --tasks
"en_US.UTF-8"
rake docker:build         # Run docker build
rake docker:push          # Run docker push
rake docker:run           # Run docker run without command
rake dockerfile:generate  # Generate Dockerfile from a template
rake dockerfile:verify    # Verify Dockerfile is committed
rake release[version]     # Perform a release work
```

**Comment by gnakagaw** [comment]
@ybiquitous Thanks. Current my development environment is ok.

**Comment by ybiquitous** [comment]
@gnakagaw Thank you so much! 😄

**Pull request merged**